doc/introspection.rst

```diff
@@ -0,0 +1,81 @@
+
+Introspection of the ros2_control setup
+***************************************
+
+With the integration of the ``pal_statistics`` package, the ``controller_manager`` node publishes the registered variables within the same process to the ``~/introspection_data`` topics.
+By default, all ``State`` and ``Command`` interfaces in the ``controller_manager`` are registered when they are added, and are unregistered when they are removed from the ``ResourceManager``.
+The state of the all the registered entities are published at the end of every ``update`` cycle of the ``controller_manager``. For instance, In a complete synchronous ros2_control setup (with synchronous controllers and hardware components), this data in the ``Command`` interface is the command used by the hardware components to command the hardware.
```

Comment:
As we get more controllers that support variants do you see any issues with this approach and the values not always being doubles? For example lets say the `Command` interface is a complete trajectory for the robot to execute. How would pal_statistics handle this?

Reply:
You can write a custom includes to be able to introspect custom data as well. It is documented in docs. 

However, with variants right now it is limited to double types so the use can do what he wants